docs/source/format/CanonicalExtensions.rst

```diff
@@ -0,0 +1,67 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _format_canonical_extensions:
+
+*************************
+Canonical Extension Types
+*************************
+
+============
+Introduction
+============
+
+The Arrow Columnar Format allows defining
+:ref:`extension types <format_metadata_extension_types>` so as to extend
+standard Arrow data types with custom semantics.  Often these semantics
+will be specific to a system or application.  However, it is beneficial
+to share the definitions of well-known extension types so as to improve
+interoperability between different systems integrating Arrow columnar data.
+
+Procedures
+==========
+
+These rules must be followed for the standardization of canonical extension
+types:
+
+* Canonical extension types are described and maintained below in this document.
+
+* Each canonical extension type requires a distinct discussion and vote
+  on the `Arrow development mailing-list <https://arrow.apache.org/community/>`__.
+
+* The specification text to be added *must* follow these requirements:
+
+  1) It *must* define a well-defined extension name starting with "``arrow.``".
+
+  2) Its parameters, if any, *must* be described in the proposal.
+
+  3) Its serialization *must* be described in the proposal and should
+     not require unduly work or unusual software dependencies (for example, a
```

Comment:
What do you mean by "unduly work"? Is that in terms of readability to reviewers? Or in terms of computation time?

Reply:
I meant in terms of implementation effort.

Reply:
I can make this more explicit, though ideally we should not deviate too much from what has been voted on on the ML.